libafl/src/executors/inprocess/inner.rs

```diff
@@ -65,36 +66,62 @@ impl<HT, I, OT, S> HasObservers for GenericInProcessExecutorInner<HT, I, OT, S>
     }
 }
 
-impl<HT, I, OT, S> GenericInProcessExecutorInner<HT, I, OT, S>
-where
-    OT: ObserversTuple<I, S>,
+/// A simple guard, to be used with [`GenericInProcessExecutorInner`].
+#[derive(Debug)]
+pub struct InProcessGuard<'a> {
```

Comment:
Shouldn't/coudn't this guy have the type of the generics used? As it is now you can still use it to call the methods with a different type, right?

Reply:
You would always be able to call the methods with a different type, but there are other ways that this is restricted (i.e. the harness gets moved out of the executor when it's in use, then restored after). There is no real way to restrict this further, without holding EM/Z in phantom data.